cmd/main.go

```diff
@@ -132,19 +133,22 @@ func main() {
 		r.InitializeMetricsHandler(options.HttpEndpoint, "/metrics")
 	}
 
+	cfg := metadata.MetadataServiceConfig{
+		EC2MetadataClient: metadata.DefaultEC2MetadataClient,
+		K8sAPIClient:      metadata.DefaultKubernetesAPIClient,
+	}
+
 	region := os.Getenv("AWS_REGION")
+
+	md, metadataErr := metadata.NewMetadataService(cfg, region)
+	if metadataErr != nil {
+		klog.ErrorS(err, "Could not determine region from any metadata service. The region can be manually supplied via the AWS_REGION environment variable.")
+		panic(err)
+	}
```

Comment:
Why is this moved? Moving it out of the `if region == ""` statement breaks the ability to set the region via `AWS_REGION` when metadata is not available.

Reply:
The node service heavily relies on metadata.

As an example, see NodeGetInfo:
- `zone := d.metadata.GetAvailabilityZone()`
- `outpostArn := d.metadata.GetOutpostArn()`
-  `d.metadata.GetInstanceID()`

It was moved out of that conditional so that it may be passed into `NewDriver`.

Reply:
The code needs to be reworked so you can pass `AWS_REGION` when running the controller then, as that is supported today and we likely have at least a few users using it

Reply:
Updated